### DISCUSSION TITLE
How to use streamlinkW.exe to run stream from a single command line ?

### DISCUSSION THREAD

**Discussion by SkYNewZ** [streamlink/streamlink#4324] [Q&A]
Hello.

In the same logic as https://github.com/streamlink/streamlink-twitch-gui/discussions/768 I'm trying to write a simple .bat file from which I can launch predefined stream from a simple click.
While setting Streamlink GUI in debug mode, I can get the used command:

```shell
C:\\Program Files (x86)\\Streamlink\\bin\\streamlinkw.exe "--twitch-disable-hosting" "--twitch-disable-ads" "--twitch-low-latency" "--player" "C:\\Program Files\\VideoLAN\\VLC\\vlc.exe" "--player-args" "--play-and-exit --qt-minimal-view" "--hls-live-edge" "3" "--stream-segment-threads" "1" "--retry-open" "3" "--retry-streams" "2" "twitch.tv/xx" "best"
```

But If I copy paste this exact command into my script file (or in a PowerShell terminal), nothing happens.
Even a simple command do not show anything:

```shell
PS C:\Program Files (x86)\Streamlink\bin> .\streamlinkw.exe --version
PS C:\Program Files (x86)\Streamlink\bin>
```

Can you help me find the right order, please?

**Comment by bastimeyer** [comment]
Wrong repo. You should ask this on the streamlink repo, not Streamlink Twitch GUI. Let me answer anyway.

Which version of Streamlink did you install? The 3.1.0 release had an issue with the executables. See Streamlink's 3.1.1 release.
https://github.com/streamlink/streamlink/releases/tag/3.1.1

The executables with the w suffix (pythonw.exe or streamlinkw.exe) are meant to be launched from GUI applications, because they don't force a terminal window on Windows. They also do something weird with the output streams, so if you just want to execute streamlink from the command line shell, either in interactive mode or from a script, just use the executable without the w suffix.

**Reply by SkYNewZ** [reply]
Thanks for your quick reply.

Sorry I'm here because I want to do the exact same job as Streamlink GUI. Indeed, I want to run streams from my Streamdeck, with a simple key and I was debugging with a command line/script file.

I have the latest version (in the cmd.exe)
```
C:\Program Files (x86)\Streamlink\bin>streamlink.exe --version
streamlink 3.1.1

C:\Program Files (x86)\Streamlink\bin>streamlinkw.exe --version

C:\Program Files (x86)\Streamlink\bin>echo %errorlevel%
0

```

Yes I exactly don't want "a terminal window on Windows" and here are my available options:
![image](https://user-images.githubusercontent.com/26750012/152029374-c50e1845-b3df-4faf-b72d-333af6d10451.png)

**Reply by bastimeyer** [reply]
I've moved the discussion to the Streamlink repo now, as it doesn't belong on the Twitch GUI repo.

Did you install Streamlink via python-pip (version 22)?

Does launching the stream work with the exact same parameters defined in your application when using `streamlink.exe` instead of `streamlinkw.exe` (so the only difference being the executable name)?

**Reply by SkYNewZ** [reply]
I have installed Streamlink from the releases page https://github.com/streamlink/streamlink/releases/download/3.1.1/streamlink-3.1.1.exe.

Yes. Just nothing appear using the wrapped binary.
```
C:\Program Files (x86)\Streamlink\bin>streamlinkw.exe --version

C:\Program Files (x86)\Streamlink\bin>streamlink.exe --stream-url twitch.tv/xewtwo best
https://video-weaver.mil02.hls.ttvnw.net/v1/playlist/CpoEmu4l6PZi97Fjpk8ts0lkpnT7XuIvdBas6kYDE5MuQxkGq5-VmVxiR_Sn3RkzTWP2vQUVd7dmzOaNE1fDWYhhuNS2BRFOAnH9rh-XbdhjnovP-6B4lTgHKsehb5YMNA76m2JlGYMu8iB_9dncU-fJIGZ7f0dixVoiCtiXHNeiJe-X7LI3oeihvS7AQNTWCazzpztarmbcr8IxS1wnJy9e6MTXcseHu7aEkYeJNS15E66wsPqPn_xX8JEUEXUeyWwsJkayeWF6hkEJ-CPmZjuvyk9mtnGbybMWxIMtqp4LYjmZjwW_m2MAu_6u9WORdLg55XgnFhh5aW89KkxtEkXb_7J_9gRN5P1il0BVStfe4Vv5_n8QjODGmgRGdOTJCMTcboNlSSEvmkNaXv4e3BLwZfFXWH9pstYONxbI89hnIvvqcvIjU_b9ZPBatpIR3AUy02mWKhHXFbihBfHEueqH7oTNIc8w_mgczYIAZ9f2K8PejQLXath7AxMvmpSX1SpZu0CF35VhufweXiO4GdZsEtXTdEy_0TS_eCgb-DRF3U02sMl3J1_drrNQG0ARTiDwLF9Om3Y27-kRx9l71RNPZcCa1Rl2kaU7HtfDF2y8OBxFDxrzaAQDB5jUYaV0mmqp8R0YfkG-oD2OxayltsNyRnxi1IEgPl6DMfD0j961oEQAJCen-KhXqhGzejToMC7HeOZfA2i53r2UYxoM6HFnssiASujvH630IAEqCWV1LXdlc3QtMjD1Ag.m3u8

C:\Program Files (x86)\Streamlink\bin>streamlinkw.exe --stream-url twitch.tv/xewtwo best

```

**Reply by bastimeyer** [reply]
As I've said, the w-suffix executables are not meant to be executed from the command line shell on Windows. They are meant for being executed directly from GUI applications. Use the non-w-suffix executables.

This works exactly the same as with `python.exe` and `pythonw.exe`.

**Reply by SkYNewZ** [reply]
Sorry for the misunderstanding. I'm trying to run the whole thing with only the binary path and run arguments as parameters (see my screenshot above).

So according to what you tell me, I can't do without this command prompt that appears?

The purpose of my question here is precisely how to launch streamlink without this command prompt.

**Reply by bastimeyer** [reply]
You've been posting two different things. Execution from a command line shell and execution from a GUI application. I don't know the GUI application you have posted, so I don't know what exactly it does when invoking the executable, and it's possible that it wraps your input in another shell script and the executable with the w suffix therefore doesn't work. This is however out of scope of this project and this issue tracker / discussion forum.

Please read up on the distinction between python.exe and pythonw.exe on Windows. The two streamlink executables are just wrappers for this and they are based on distlib 0.3.3. As I said, the terminal window isn't the only distinction with these executables and output streams are handled differently, too.
https://stackoverflow.com/questions/9705982/pythonw-exe-or-python-exe

Another thing you can try doing is invoking `python.exe` or `pythonw.exe` directly and passing the streamlink executable(s) as the first argument with any streamlink arguments as further parameters.

If this doesn't help you, then you have to ask on the support channels of the GUI application you're using where you're trying to execute streamlink.

**Reply by SkYNewZ** [reply]
Thanks for your replies and your help.
